tools/payments/client.go

```diff
@@ -0,0 +1,210 @@
+package payments
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/brave-intl/bat-go/libs/httpsignature"
+	"github.com/google/uuid"
+	"github.com/redis/go-redis/v9"
+)
+
+const (
+	preparePrefix = "prepare-"
+	submitPrefix  = "authorize-"
+
+	// headers
+	hostHeader          = "Host"
+	digestHeader        = "Digest"
+	contentLengthHeader = "Content-Length"
+	contentTypeHeader   = "Content-type"
+	signatureHeader     = "Signature"
+)
+
+var (
+	payout        = strconv.FormatInt(time.Now().Unix(), 10)
+	prepareStream = preparePrefix + payout
+	submitStream  = preparePrefix + payout
+
+	prepareConfigStream = preparePrefix + "-configure"
```

Comment:
Will this end up with `prepare-configurer` i.e. two hyphens. 

The same for submit